### PR TITLE
Clean up the parsers into a single interface

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/13675)

### CODE REVIEW

File: components/script/dom/servoparser.rs [hunk 1]
```diff
@@ -11,17 +12,37 @@ pub struct ServoParser {
...
+    pub fn push_input_chunk(&self, chunk: String) {
+        self.pending_input.borrow_mut().push(chunk);
+    }
+
+    pub fn take_next_input_chunk(&self) -> Option<String> {
+        let mut pending_input = self.pending_input.borrow_mut();
+        if pending_input.is_empty() {
+            None
+        } else {
+            Some(pending_input.remove(0))
```
Comment (Ms2ger) [inline components/script/dom/servoparser.rs]: Followup: should use `pop()` (and probably a queue).

File: components/script/dom/servohtmlparser.rs [hunk 2]
```diff
@@ -268,10 +269,9 @@ impl ServoHTMLParser {
         let tok = tokenizer::Tokenizer::new(tb, Default::default());
 
         let parser = ServoHTMLParser {
-            servoparser: ServoParser::new_inherited(document),
+            servoparser: ServoParser::new_inherited(document, false),
```
Comment (Ms2ger) [inline components/script/dom/servohtmlparser.rs]: The boolean `last_chunk_received` argument. I'd prefer an isomorphic enum.

**Review by Ms2ger — APPROVED** [review]
I'll assume you can c/p.

**Comment by nox** [comment]
@Ms2ger Made the change.

**Pull request merged**